BUG:  chokes on pd.DatetimeTZDtype if there are no rows. (#59123)
pandas-dev/pandas: doc/source/whatsnew/v3.0.0.rst

^^^^^^^^^
- Bug in :meth:`DataFrame.join` inconsistently setting result index name (:issue:`55815`)
- Bug in :meth:`DataFrame.unstack` producing incorrect results when ``sort=False`` (:issue:`54987`, :issue:`55516`)

> - Bug in :meth:`DataFrame.unstack` producing incorrect results when manipulating empty DataFrame using dtype extensions (:issue:`59123`)

Reviewer: ```suggestion
- Bug in :meth:`DataFrame.unstack` producing incorrect results when manipulating empty :class:`DataFrame` with an :class:`ExtentionDtype` (:issue:`59123`)
```

Author: Thanks. Done.
